android/src/main/java/com/rnziparchive/RNZipArchiveModule.java

```diff
@@ -326,6 +328,97 @@ public void zip(String fileOrDirectory, String destDirectory, Promise promise) {
     }
   }
 
+  @ReactMethod
+  public void zipWithPassword(final String fileOrDirectory, final String destDirectory, final String password,
+      final String encyptionMethod, final Promise promise) {
+    new Thread(new Runnable() {
+      @Override
+      public void run() {
+
+        try {
+          net.lingala.zip4j.core.ZipFile zipFile = new net.lingala.zip4j.core.ZipFile(destDirectory);
+
+          ZipParameters parameters = new ZipParameters();
+          parameters.setCompressionMethod(Zip4jConstants.COMP_DEFLATE);
+          parameters.setCompressionLevel(Zip4jConstants.DEFLATE_LEVEL_NORMAL);
+
+          String encParts[] = encyptionMethod.split("-");
+
+          if (password != null && !password.isEmpty()) {
+            parameters.setEncryptFiles(true);
+            if (encParts[0].equals("AES")) {
+              parameters.setEncryptionMethod(Zip4jConstants.ENC_METHOD_AES);
+              if (encParts[1].equals("128")) {
+                parameters.setAesKeyStrength(Zip4jConstants.AES_STRENGTH_128);
+              } else if (encParts[1].equals("256")) {
+                parameters.setAesKeyStrength(Zip4jConstants.AES_STRENGTH_256);
+              } else {
+                parameters.setAesKeyStrength(Zip4jConstants.ENC_METHOD_STANDARD);
+              }
+            } else if (encyptionMethod.equals("STANDARD")) {
+              parameters.setEncryptionMethod(Zip4jConstants.ENC_METHOD_STANDARD);
+              Log.d(TAG, "Standard Encryption");
+            } else {
+              parameters.setEncryptionMethod(Zip4jConstants.ENC_METHOD_STANDARD);
+              Log.d(TAG, "Encryption type not supported default to Standard Encryption");
+            }
+            parameters.setPassword(password);
+          } else {
+            promise.reject(null, "Password is empty");
+          }
+
+          updateProgress(0, 100, destDirectory);
+
+          File f = new File(fileOrDirectory);
+
+          int totalFiles = 0;          
+          int fileCounter = 0;
+          
+          if (f.exists()) {
+            if (f.isDirectory()) {
+
+              zipFile.addFolder(f.getAbsolutePath(), parameters);
+              fileCounter += 1;
+
+              List<File> files = getSubFiles(f, true);
+
+              totalFiles = files.size() + 1;
+              updateProgress(fileCounter, totalFiles, destDirectory);
+              for (int i = 0; i < files.size(); i++) {
+                if (files.get(i).isDirectory()) {
+                  zipFile.addFolder(f.getAbsolutePath(), parameters);
```

Comment:
I think this call should be 
```suggestion
                  zipFile.addFolder(files.get(i).getAbsolutePath(), parameters);
```
, not tested out but seems more reasonable in this way.